packages/mui-joy/src/styles/extendTheme.ts

```diff
@@ -644,5 +645,38 @@ export default function extendTheme(themeOptions?: CssVarsThemeOptions): Theme {
     });
   };
 
+  // @ts-ignore palette could be defined in the ThemeProvider's theme
+  theme.palette = deepmerge(themeOptions?.palette || {}, {
+    ...(theme.colorSchemes.light.palette as RuntimeColorSystem['palette']),
+    colorScheme: 'light',
+  });
+  theme.vars = {
+    focus: theme.focus,
+    fontFamily: theme.fontFamily,
+    fontSize: theme.fontSize,
+    fontWeight: theme.fontWeight,
+    letterSpacing: theme.letterSpacing,
+    lineHeight: theme.lineHeight,
+    radius: theme.radius,
+    shadow: theme.shadow,
+    palette: theme.palette,
+    shadowRing: theme.shadowRing,
+    shadowChannel: theme.shadowChannel,
+  };
+  theme.getColorSchemeSelector = () => '&';
+
+  // @ts-ignore if the colorInversion is provided as callbacks, it needs to be resolved in the CssVarsProvider
+  theme.colorInversion =
+    typeof colorInversionInput === 'function'
```

Comment:
I decided not to resolve the `colorInversion` as function in the `extendTheme`, but rather leave that in the `CssVarsProvider` as some tests were failing, for e.g. the ones that depend that the `theme` in the callback, has the `getColorSchemeSelector` implementation that based on the input color scheme generates different result.